File: loops/min_max.py
maxval = None
minval = None

while True:
    val = input()
    val = val.strip()
    if val == "q": 
        break
    val = int(val)
    if maxval == None or val > maxval:
        maxval = val
    if minval == None or val < minval:
        minval = val

print('Maxval: ', maxval)

print('Minval: ', minval)

File: loops/min_max.c
#include <stdio.h>

int main()
{
	int first = 1;
	int val, maxval, minval;

	while (scanf("%d", &val) != EOF)
	{
		if (first || val < minval)
			minval = val;
		if (first || val > maxval)
			maxval = val;
		first = 0;
	}
	printf("Maxvalue: %d\n", maxval);
	printf("Minvalue: %d\n", minval);
}
